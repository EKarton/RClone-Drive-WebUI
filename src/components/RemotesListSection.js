import Typography from '@mui/material/Typography';
import RemoteCardList from 'components/RemoteCardList';
import useFetchRemotes from 'hooks/fetch-data/useFetchRemotes';
import { StatusTypes } from 'utils/constants';
import './RemotesListSection.scss';

/**
 * Represents the list of remotes with header
 */
export default function RemotesListSection({ onRemoteCardClicked }) {
  const { status, data: remotes, error } = useFetchRemotes();

  if (status === StatusTypes.LOADING) {
    return null;
  }

  if (status === StatusTypes.ERROR) {
    throw error;
  }

  const handleRemoteCardClicked = (remote) => {
    onRemoteCardClicked(remote);
  };

  return (
    <>
      <Typography
        variant="h7"
        component="div"
        color="text.primary"
        className="remotes-list-section__header-text"
      >
        Your Remotes
      </Typography>
      <RemoteCardList remotes={remotes} onClick={handleRemoteCardClicked} />
    </>
  );
}
