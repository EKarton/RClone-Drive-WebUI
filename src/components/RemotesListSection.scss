.remotes-list-section {
  &__header-text {
    padding-bottom: 1rem;
    font-weight: bold;
  }
}
