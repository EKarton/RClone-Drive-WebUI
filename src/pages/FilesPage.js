import Toolbar from "@mui/material/Toolbar";
import "./FilesPage.scss";
import GlobalAppBar from "components/GlobalAppBar";
import FolderIcon from "@mui/icons-material/Folder";
import DescriptionIcon from "@mui/icons-material/Description";
import { Table } from "semantic-ui-react";

const FilesPage = () => {
  const rows = [
    { id: 1, col1: "Hello", col2: "World", col3: "123", col4: "Folder" },
    {
      id: 2,
      col1: "DataGridPro",
      col2: "is Awesome",
      col3: "343",
      col4: "File",
    },
    { id: 3, col1: "MUI", col2: "is Amazing", col3: "123", col4: "File" },
  ];

  const columns = [
    { field: "col1", headerName: "Name", flex: 1, minWidth: 150 },
    { field: "col2", headerName: "Last Updated", width: 200 },
    { field: "col3", headerName: "File Size", flex: 0.3, minWidth: 50 },
    { field: "col4", headerName: "Kind", flex: 0.3, minWidth: 50 },
  ];

  return (
    <>
      <GlobalAppBar />
      <Toolbar />
      <Table celled striped>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="3">Git Repository</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell collapsing>
              <FolderIcon /> node_modules
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell collapsing textAlign="right">
              10 hours ago
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <FolderIcon /> test
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <FolderIcon /> build
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <DescriptionIcon /> package.json
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <DescriptionIcon /> Gruntfile.js
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    </>
  );
};

export default FilesPage;
