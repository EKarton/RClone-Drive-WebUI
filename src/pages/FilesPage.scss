.table {
  & .MuiDataGrid-cell:focus {
    outline: none !important;
  }
}
