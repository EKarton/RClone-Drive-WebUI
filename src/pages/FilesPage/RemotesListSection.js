import { useHistory } from 'react-router';
import { hashRemotePath } from 'utils/remote-paths-url';
import RemoteCardList from 'components/RemoteCardList';
import { StatusTypes } from 'utils/constants';
import useFetchRemotes from 'hooks/fetch-data/useFetchRemotes';

export default function RemotesListSection() {
  const history = useHistory();
  const { status, data: remotes, error } = useFetchRemotes();

  if (status === StatusTypes.LOADING) {
    return null;
  }

  if (status === StatusTypes.ERROR) {
    return <div>{error.message}</div>;
  }

  const handleRemoteCardClicked = (remote) => {
    history.push(`/files/${hashRemotePath(`${remote}:`)}`);
  };

  return (
    <>
      <h4>Your Remotes</h4>
      <RemoteCardList remotes={remotes} onClick={handleRemoteCardClicked} />
    </>
  );
}
