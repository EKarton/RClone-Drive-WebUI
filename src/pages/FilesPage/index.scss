.files-page {
  padding-bottom: 3rem;
}
