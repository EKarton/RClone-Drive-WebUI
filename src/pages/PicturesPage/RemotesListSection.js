import RemoteCardList from 'components/RemoteCardList';
import { StatusTypes } from 'utils/constants';
import useFetchRemotes from 'hooks/fetch-data/useFetchRemotes';

/**
 * Represents the list of remotes in the Pictures page
 */
export default function RemotesListSection({ onRemoteCardClicked }) {
  const { status, data: remotes } = useFetchRemotes();

  if (status === StatusTypes.LOADING) {
    return null;
  }

  if (status === StatusTypes.ERROR) {
    return 'Error!';
  }

  const handleRemoteCardClicked = (remote) => {
    onRemoteCardClicked(remote);
  };

  return (
    <>
      <h4>Your Remotes</h4>
      <RemoteCardList remotes={remotes} onClick={handleRemoteCardClicked} />
    </>
  );
}
