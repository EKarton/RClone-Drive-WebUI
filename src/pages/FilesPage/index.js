import { useNavigate } from 'react-router-dom';
import RemotesListSection from 'components/RemotesListSection';
import { hashRemotePath } from 'utils/remote-paths-url';
import './index.scss';

export default function FilesPage() {
  const navigate = useNavigate();

  const handleRemoteCardClicked = (remote) => {
    navigate(`/files/${hashRemotePath(`${remote}:`)}`);
  };

  return (
    <div className="files-page">
      <RemotesListSection onRemoteCardClicked={handleRemoteCardClicked} />
    </div>
  );
}
