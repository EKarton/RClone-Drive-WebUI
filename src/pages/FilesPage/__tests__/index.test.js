import { customRender } from 'test-utils/react';
import RemotesListSection from '../RemotesListSection';
import FilesPage from '..';

jest.mock('../RemotesListSection');

describe('FilesPage', () => {
  it('should match snapshot', () => {
    RemotesListSection.mockReturnValue(null);

    const { baseElement } = customRender(<FilesPage />);

    expect(baseElement).toMatchSnapshot();
  });
});
