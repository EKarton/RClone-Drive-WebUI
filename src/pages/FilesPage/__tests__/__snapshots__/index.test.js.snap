// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesPage should match snapshot 1`] = `
<body>
  <div>
    <div
      class="files-page"
    />
  </div>
</body>
`;
